stdlib/printexc.ml

```diff
@@ -61,14 +74,13 @@ let to_string_default = function
       sprintf locfmt file line char (char+6) "Assertion failed"
   | Undefined_recursive_module(file, line, char) ->
       sprintf locfmt file line char (char+6) "Undefined recursive module"
+  | CamlinternalEffect.Unhandled e ->
+      sprintf "Stdlib.Effect.Unhandled(%s)"
```

Comment:
```suggestion
      sprintf "Unhandled effect: %s"
```
Given that we are providing more user-friendly messages for other built-in exceptions, I think this would look nicer.

Reply:
Let's keep the usual exception message style for now. We can decide to improve the message at any later point (contrarily to the exception names).